cc3200-sys/simplelink.c

```diff
@@ -225,12 +225,14 @@ void SimpleLinkPingReport(SlPingReport_t *pPingReport)
     g_ulPingPacketsRecv = pPingReport->PacketsReceived;
 }
 
-_u32 sl_FsMode(bool write, bool create, _u32 maxCreateSize)
+_u32 sl_FsMode(bool write, bool create, bool failsafe, _u32 maxCreateSize)
 {
     if (create) {
-        return FS_MODE_OPEN_CREATE(maxCreateSize,
-                                   _FS_FILE_OPEN_FLAG_COMMIT |
-                                   _FS_FILE_PUBLIC_WRITE);
+        _u32 flags = 0;
+        if (failsafe) {
+            flags |= _FS_FILE_OPEN_FLAG_COMMIT;
```

Comment:
Feel free to keep it like that but you could s/|=/=
Or even just use a ternary operator.